src/main/java/com/iota/iri/network/protocol/ProtocolMessage.java

```diff
@@ -0,0 +1,87 @@
+package com.iota.iri.network.protocol;
+
+/**
+ * Defines the different message types supported by the protocol and their characteristics.
+ */
+public enum ProtocolMessage {
```

Comment:
It's not a segment of a protocol message, each enum literally defines a protocol message in its entirety. Won't rename as I don't think this makes sense.

Reply:
We can abstract and make the `MessageReader` as you want in the next version. I agree with you on this. So or so, we will end up with objects implementing a concrete protocol version which then already will be thorn away from the neighbor code into its own package. Basically you will get an object implementing the specifics about a protocol version via a factory in the end.

The current implementation is not set in stone by all means but we don't need to change it now, as it would only defer things unnecessarily. I've opened https://github.com/iotaledger/iri/issues/1498 to track this.